src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -169,18 +169,26 @@ class WorkspaceService(userInfo: UserInfo, dataSource: DataSource, containerDAO:
           if (accessLevel < WorkspaceAccessLevels.Read)
             Future.successful(RequestComplete(ErrorReport(StatusCodes.NotFound, noSuchWorkspaceMessage(workspaceName))))
           else {
-            gcsDAO.getOwners(workspaceContext.workspace.workspaceId) map { owners =>
-              val response = WorkspaceListResponse(accessLevel,
-                workspaceContext.workspace,
-                getWorkspaceSubmissionStats(workspaceContext, txn),
-                owners)
-              RequestComplete(StatusCodes.OK, response)
-            }
+            val owners = getWorkspaceOwners(workspaceName).value.get.get
+            val response = WorkspaceListResponse(accessLevel,
+              workspaceContext.workspace,
+              getWorkspaceSubmissionStats(workspaceContext, txn),
+              owners)
+            Future.successful(RequestComplete(StatusCodes.OK, response))
           }
         }
       }
     }
 
+  def getWorkspaceOwners(workspaceName: WorkspaceName): Future[Seq[String]] =
+    dataSource.inTransaction(readLocks=Set(workspaceName)) { txn =>
+      val ownerGroup = containerDAO.authDAO.loadGroup(UserAuth.toWorkspaceAccessGroupName(workspaceName, WorkspaceAccessLevels.Owner), txn)
+      val users = ownerGroup.users.map(u => gcsDAO.toUserFromProxy(gcsDAO.toProxyFromUser(u.userSubjectId)))
+      val subGroups = ownerGroup.subGroups.map(g => g.groupName)
+
+      Future.successful((users++subGroups).toSeq)
```

Comment:
I think instead of this you want to change `inTransaction` to `inFutureTransaction` and put the contents in a `Future {}` block, but see what doge says